### ISSUE TITLE
Cannot generate an APK for internal distribution

### ISSUE THREAD

**Issue by bradjones1** [expo/eas-cli#1506]
### Build/Submit details page URL

https://expo.dev/accounts/meet-kinksters/projects/kinksters/builds/9e0226c4-fe36-45ca-8ba3-eb44b6e67890

### Summary

I am attempting to generate an APK that is directly installable/side-loaded on Android, however every build I submit to EAS Build with the below `production-sideload` configuration yields an AAB.

### Managed or bare?

Prebuild

### Environment

```
$ npx expo-env-info

  expo-env-info 1.0.3 environment info:
    System:
      OS: Linux 5.18 Debian GNU/Linux 10 (buster) 10 (buster)
      Shell: 5.0.3 - /bin/bash
    Binaries:
      Node: 16.13.1 - /usr/local/bin/node
      Yarn: 1.22.15 - /usr/local/bin/yarn
      npm: 8.1.2 - /usr/local/bin/npm
    npmPackages:
      expo: ^45.0.0 => 45.0.5 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

```
{
  "build": {
    "production-sideload": {
      "extends": "production",
      "distribution": "internal",
      "android": {
        "buildType": "apk"
      }
    },
    "production": {
      "autoIncrement": true,
      "channel": "production",
      "distribution": "store",
      "env": {
        "FLIPPER_DISABLE": "1",
        "SENTRY_ORG": "not-vanilla-llc",
        "SENTRY_PROJECT": "react-native"
      }
    }
  },
  "cli": {
    "version": ">=2.6.0",
    "appVersionSource": "remote"
  }
}
```

I am curious if this is related to my having an `app.config.js` file instead of static configuration. Here is a slightly sanitized version:

```
const isDevelopmentProfile = !process.env.EAS_BUILD_PROFILE.startsWith('production');

const packageSchemeName = isDevelopmentProfile
  ? "dating.kinksters.meet.dev"
  : "dating.kinksters.meet";

module.exports = {
  "expo": {
    "name": isDevelopmentProfile ? "Dev - Meet Kinksters" : "Meet Kinksters",
    "version": "1.0.11",
    "slug": "kinksters",
    "owner": "meet-kinksters",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "jsEngine": "hermes",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 10000,
      "url": "https://u.expo.dev/4c2a39d8-bd5a-43b9-a206-826888278dcc"
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "bundleIdentifier": packageSchemeName,
      "supportsTablet": true,
      "infoPlist": {
        "SKAdNetworkItems": [
          {
            "SKAdNetworkIdentifier": "x"
          },
          {
            "SKAdNetworkIdentifier": "y"
          }
        ]
      }
    },
    "android": {
      "package": packageSchemeName,
      "googleServicesFile": "./google-services.json",
      "permissions": [
        "android.permission.INTERNET"
      ],
    },
    "scheme": packageSchemeName,
    "plugins": [
      [
        "expo-community-flipper",
        "0.123.0"
      ],
      [
        "react-native-fbsdk-next",
        {
          "appID": isDevelopmentProfile ? "x" : "y",
          "displayName": "Meet Kinksters",
          "clientToken": isDevelopmentProfile ? "x" : "y"
        }
      ],
      ["sentry-expo"]
    ],
    "extra": {
      "eas": {
        "projectId": "4c2a39d8-bd5a-43b9-a206-826888278dcc"
      }
    },
    "runtimeVersion": "1.0.0"
  },
  "hooks": {
    "postPublish": [
      {
        "file": "sentry-expo/upload-sourcemaps"
      }
    ]
  }
}
```

Build command:

`EAS_BUILD_PROFILE=production-sideload eas build --platform android --non-interactive`

**Comment by bradjones1** [comment]
This feels more or less the same as https://github.com/expo/eas-cli/pull/349 but that was marked as fixed in 0.11 so I'm not sure what might be the problem, here.

**Comment by bradjones1** [comment]
Ugh, this is PEBKAC. I am setting the environment variable to the profile I want to build so that `app.config.js` will be properly evaluated, but neglected that `eas build` doesn't read from that, so I needed to explicitly set `--profile` in the command invocation, as well.
